### PR TITLE
MINIFICPP-617: Create simple python example

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: python/getFile.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+parser.add_argument("-i", "--input", dest="input_port",
+                    help="NiFi Input Port")
+                    
+parser.add_argument("-d", "--dir", dest="dir",
+        	help="GetFile Dir to monitor", metavar="FILE")
+                    
+args = parser.parse_args()
+
+""" dll_file is the path to the shared object """
+minifi = MiNiFi(dll_file=args.dll_file,url = args.nifi_instance.encode('utf-8'), port=args.input_port.encode('utf-8'))
```
Comment (phrocker) [inline python/getFile.py]: probably no need to do the encoding here

**Comment by apiri** [comment]
reviewing

File: libminifi/src/capi/Plan.cpp [hunk 2]
```diff
@@ -35,6 +36,41 @@ ExecutionPlan::ExecutionPlan(std::shared_ptr<core::ContentRepository> content_re
...
+  if (finalized) {
+    return nullptr;
+  }
+
+  utils::Identifier uuid;
+  id_generator_->generate(uuid);
+
+  auto ptr = core::ClassLoader::getDefaultClassLoader().instantiate("CallbackProcessor", uuid);
+  if (nullptr == ptr) {
+    throw std::exception();
```
[the remark this answers is not in the document]
Comment (phrocker) [reply libminifi/src/capi/Plan.cpp]: I agree we should probably do this. At the time this was created there was no discussion of that type of work -- happy to make a follow on ticket -- and I have -- https://issues.apache.org/jira/browse/MINIFICPP-639

**Comment by apiri** [comment]
reviewing latest update

File: PYTHON.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+is routed to success. Custom routes can be defined in later implementations of the Python API.
+
+An RPG is currently required to define a MiNiFi instance. As per the example, a flow file may 
+be transmitted via HTTP site to site. Presently, raw socket site to site is supported via
+the CAPI but not the Python library.
+
+To run the getfile example you will need to provide the fullpath to the python-lib shared object
+created at build time.
+
+## Enabling
```
Comment (apiri) [inline PYTHON.md]: Verified functionality against python2.7 and python3.  I did not have cffi installed for python3 but did for python 2.7.  We should probably include that as a requirement.  Are there any other requirements we should call out?

Comment (phrocker) [reply PYTHON.md]: Ah yeah good i forgot that. thanks!

Comment (phrocker) [reply PYTHON.md]: Sorry I didn't have to install that on my machine ( or maybe I had done it long ago) Thanks for the reminder

Comment (apiri) [reply PYTHON.md]: Looks good, thanks for the updates.  Will get this merged in.

**Pull request closed without merging**